engine/classes/Elgg/Filesystem/AdapterService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+namespace Elgg\Filesystem;
+use Elgg\Filesystem\Adapter\Adapter;
+
+
+class AdapterService {
+	private $logger;
+	
+	/**
+	 *
+	 * @var string
+	 */
+	private $default;
+	
+	/**
+	 *
+	 * @var Adapter[]
+	 */
+	private $adapters;
+	
+	public function __construct(\Elgg\Logger $logger) {
+		$this->logger = $logger;
+	}
+	
+	public function set($name, Adapter $adapter, $default = false) {
+		if ($this->has($name)) {
+			throw new \UnexpectedValueException("Filesystem adapter \"$name\" already exists.");
+		}
+		$this->adapters[$name] = $adapter;
+		
+		if ($default) {
+			$this->setDefault($name);
+		}
+		return true;
+	}
+	
+	/**
+	 * 
+	 * @param type $name
+	 * @return Adapter
+	 * @throws \InvalidArgumentException
+	 */
+	public function get($name) {
+		if (!$this->has($name)) {
+			throw new \InvalidArgumentException("Filesystem Adapter \"$name\" does not exist.");
+		}
+		
+		return $this->adapters[$name];
+	}
+	
+	public function has($name) {
+		return isset($this->adapters[$name]);
+	}
+	
+	public function remove($name) {
```

Comment:
Which ones? The only one I see that takes a value is ElggPriorityList. All the other ones take ->remove($key) or ->delete($key). It's not a huge deal since there needs to be documentation, but I'd rather follow current convention than introduce more inconsistent behavior.